components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/ClientAuthenticationHandler.java

```diff
@@ -78,6 +81,10 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) thr
         AuthenticationContext authenticationContext = (AuthenticationContext) messageContext;
         String authorizationHeader = authenticationContext.getAuthenticationRequest().
                 getHeader(HttpHeaders.AUTHORIZATION);
+        String serviceProvider = authenticationContext.getAuthenticationRequest().getHeader(SERVICE_PROVIDER);
```

Comment:
Better to remove line 51 constant and use SERVICE_PROVIDER_KEY.toLowerCase() here to avoid confusion.

Reply:
Fixed with https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/129